In [1]:
import pandas as pd

df = pd.read_csv(r"C:\Users\USER\OneDrive\Desktop\Business_Analysis_Agent\data\raw\zomato.csv")

print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

print("\nColumns:")
print(df.columns.tolist())

print("\nFirst 5 rows:")
display(df.head())

print("\nData types:")
print(df.dtypes)

Rows: 51717
Columns: 17

Columns:
['url', 'address', 'name', 'online_order', 'book_table', 'rate', 'votes', 'phone', 'location', 'rest_type', 'dish_liked', 'cuisines', 'approx_cost(for two people)', 'reviews_list', 'menu_item', 'listed_in(type)', 'listed_in(city)']

First 5 rows:


,url,address,name,online_order,book_table,rate,votes,phone,location,rest_type,dish_liked,cuisines,approx_cost(for two people),reviews_list,menu_item,listed_in(type),listed_in(city)
0,https://www.zomato.com/bangalore/jalsa-banasha...,"942, 21st Main Road, 2nd Stage, Banashankari, ...",Jalsa,Yes,Yes,4.1/5,775,080 42297555\r\n+91 9743772233,Banashankari,Casual Dining,"Pasta, Lunch Buffet, Masala Papad, Paneer Laja...","North Indian, Mughlai, Chinese",800,"[('Rated 4.0', 'RATED\n A beautiful place to ...",[],Buffet,Banashankari
1,https://www.zomato.com/bangalore/spice-elephan...,"2nd Floor, 80 Feet Road, Near Big Bazaar, 6th ...",Spice Elephant,Yes,No,4.1/5,787,080 41714161,Banashankari,Casual Dining,"Momos, Lunch Buffet, Chocolate Nirvana, Thai G...","Chinese, North Indian, Thai",800,"[('Rated 4.0', 'RATED\n Had been here for din...",[],Buffet,Banashankari
2,https://www.zomato.com/SanchurroBangalore?cont...,"1112, Next to KIMS Medical College, 17th Cross...",San Churro Cafe,Yes,No,3.8/5,918,+91 9663487993,Banashankari,"Cafe, Casual Dining","Churros, Cannelloni, Minestrone Soup, Hot Choc...","Cafe, Mexican, Italian",800,"[('Rated 3.0', ""RATED\n Ambience is not that ...",[],Buffet,Banashankari
3,https://www.zomato.com/bangalore/addhuri-udupi...,"1st Floor, Annakuteera, 3rd Stage, Banashankar...",Addhuri Udupi Bhojana,No,No,3.7/5,88,+91 9620009302,Banashankari,Quick Bites,Masala Dosa,"South Indian, North Indian",300,"[('Rated 4.0', ""RATED\n Great food and proper...",[],Buffet,Banashankari
4,https://www.zomato.com/bangalore/grand-village...,"10, 3rd Floor, Lakshmi Associates, Gandhi Baza...",Grand Village,No,No,3.8/5,166,+91 8026612447\r\n+91 9901210005,Basavanagudi,Casual Dining,"Panipuri, Gol Gappe","North Indian, Rajasthani",600,"[('Rated 4.0', 'RATED\n Very good restaurant ...",[],Buffet,Banashankari



Data types:
url                              str
address                          str
name                             str
online_order                     str
book_table                       str
rate                             str
votes                          int64
phone                            str
location                         str
rest_type                        str
dish_liked                       str
cuisines                         str
approx_cost(for two people)      str
reviews_list                     str
menu_item                        str
listed_in(type)                  str
listed_in(city)                  str
dtype: object


In [2]:
print(df.info())

<class 'pandas.DataFrame'>
RangeIndex: 51717 entries, 0 to 51716
Data columns (total 17 columns):
 #   Column                       Non-Null Count  Dtype
---  ------                       --------------  -----
 0   url                          51717 non-null  str  
 1   address                      51717 non-null  str  
 2   name                         51717 non-null  str  
 3   online_order                 51717 non-null  str  
 4   book_table                   51717 non-null  str  
 5   rate                         43942 non-null  str  
 6   votes                        51717 non-null  int64
 7   phone                        50509 non-null  str  
 8   location                     51696 non-null  str  
 9   rest_type                    51490 non-null  str  
 10  dish_liked                   23639 non-null  str  
 11  cuisines                     51672 non-null  str  
 12  approx_cost(for two people)  51371 non-null  str  
 13  reviews_list                 51717 non-null  str  
 14  m

In [3]:
print("\nMissing values:")
print(df.isnull().sum())


Missing values:
url                                0
address                            0
name                               0
online_order                       0
book_table                         0
rate                            7775
votes                              0
phone                           1208
location                          21
rest_type                        227
dish_liked                     28078
cuisines                          45
approx_cost(for two people)      346
reviews_list                       0
menu_item                          0
listed_in(type)                    0
listed_in(city)                    0
dtype: int64


In [4]:
print("\nDuplicate rows:")
print(df.duplicated().sum())


Duplicate rows:
0


In [5]:
categorical_cols = [
    "online_order",
    "book_table",
    "location",
    "rest_type",
    "cuisines",
    "listed_in(type)",
    "listed_in(city)"
]

for col in categorical_cols:
    print(f"\n{'='*60}")
    print(col)
    print("Unique values:", df[col].nunique())
    print(df[col].value_counts().head(15))


online_order
Unique values: 2
online_order
Yes    30444
No     21273
Name: count, dtype: int64

book_table
Unique values: 2
book_table
No     45268
Yes     6449
Name: count, dtype: int64

location
Unique values: 93
location
BTM                      5124
HSR                      2523
Koramangala 5th Block    2504
JP Nagar                 2235
Whitefield               2144
Indiranagar              2083
Jayanagar                1926
Marathahalli             1846
Bannerghatta Road        1630
Bellandur                1286
Electronic City          1258
Koramangala 1st Block    1238
Brigade Road             1218
Koramangala 7th Block    1181
Koramangala 6th Block    1156
Name: count, dtype: int64

rest_type
Unique values: 93
rest_type
Quick Bites           19132
Casual Dining         10330
Cafe                   3732
Delivery               2604
Dessert Parlor         2263
Takeaway, Delivery     2037
Casual Dining, Bar     1154
Bakery                 1141
Beverage Shop           867
Bar     

In [6]:
print(df[[
    "rate",
    "votes",
    "approx_cost(for two people)"
]].describe(include="all"))

         rate         votes approx_cost(for two people)
count   43942  51717.000000                       51371
unique     64           NaN                          70
top       NEW           NaN                         300
freq     2208           NaN                        7576
mean      NaN    283.697527                         NaN
std       NaN    803.838853                         NaN
min       NaN      0.000000                         NaN
25%       NaN      7.000000                         NaN
50%       NaN     41.000000                         NaN
75%       NaN    198.000000                         NaN
max       NaN  16832.000000                         NaN


In [7]:
print("Rate values:")
print(df["rate"].value_counts(dropna=False).head(30))

Rate values:
rate
NaN       7775
NEW       2208
3.9/5     2098
3.8/5     2022
3.7/5     2011
3.9 /5    1874
3.8 /5    1851
3.7 /5    1810
3.6/5     1773
4.0/5     1609
4.0 /5    1574
3.6 /5    1543
4.1/5     1474
4.1 /5    1474
3.5/5     1431
3.5 /5    1353
3.4/5     1259
3.4 /5    1217
3.3/5     1168
4.2 /5    1165
3.3 /5    1142
4.2/5     1019
3.2/5     1006
4.3 /5     917
3.2 /5     867
3.1/5      862
4.3/5      776
3.1 /5     699
4.4 /5     628
3.0/5      558
Name: count, dtype: int64


In [8]:
print(df["rest_type"].value_counts(dropna=False))

rest_type
Quick Bites                   19132
Casual Dining                 10330
Cafe                           3732
Delivery                       2604
Dessert Parlor                 2263
                              ...  
Pop Up                            2
Food Court, Beverage Shop         2
Dessert Parlor, Food Court        2
Sweet Shop, Dessert Parlor        1
Quick Bites, Kiosk                1
Name: count, Length: 94, dtype: int64


In [9]:
print(df["listed_in(type)"].value_counts())

listed_in(type)
Delivery              25942
Dine-out              17779
Desserts               3593
Cafes                  1723
Drinks & nightlife     1101
Buffet                  882
Pubs and bars           697
Name: count, dtype: int64


In [10]:
print(df["cuisines"].value_counts().head(30))

cuisines
North Indian                           2913
North Indian, Chinese                  2385
South Indian                           1828
Biryani                                 918
Bakery, Desserts                        911
Fast Food                               803
Desserts                                766
Cafe                                    756
South Indian, North Indian, Chinese     726
Bakery                                  651
Chinese                                 556
Ice Cream, Desserts                     417
Chinese, North Indian                   415
Mithai, Street Food                     372
Desserts, Ice Cream                     354
North Indian, Chinese, Biryani          352
South Indian, North Indian              343
North Indian, South Indian              343
North Indian, South Indian, Chinese     305
Beverages                               301
North Indian, Biryani                   285
Biryani, Kebab                          279
Finger Food            

In [11]:
print(df["location"].value_counts().head(30))

location
BTM                      5124
HSR                      2523
Koramangala 5th Block    2504
JP Nagar                 2235
Whitefield               2144
Indiranagar              2083
Jayanagar                1926
Marathahalli             1846
Bannerghatta Road        1630
Bellandur                1286
Electronic City          1258
Koramangala 1st Block    1238
Brigade Road             1218
Koramangala 7th Block    1181
Koramangala 6th Block    1156
Sarjapur Road            1065
Ulsoor                   1023
Koramangala 4th Block    1017
MG Road                   918
Banashankari              906
Kalyan Nagar              853
Richmond Road             812
Frazer Town               727
Malleshwaram              725
Basavanagudi              684
Residency Road            675
Banaswadi                 664
Brookefield               658
New BEL Road              649
Kammanahalli              648
Name: count, dtype: int64


In [12]:
print(df["approx_cost(for two people)"].value_counts().head(30))

approx_cost(for two people)
300      7576
400      6562
500      4980
200      4857
600      3714
250      2959
800      2285
150      2066
700      1948
350      1763
1,000    1637
450      1417
100       993
1,200     993
1,500     971
650       776
550       761
750       758
900       700
1,300     516
1,100     512
1,400     473
2,000     363
1,600     266
1,700     247
1,800     203
850       166
3,000     162
2,500     146
2,200      78
Name: count, dtype: int64


In [13]:
print("Missing:", df["rate"].isna().sum())
print("NEW:", (df["rate"] == "NEW").sum())

rating_mask = (
    df["rate"].notna() &
    (df["rate"] != "NEW")
)

print("Actual rating entries:", rating_mask.sum())

Missing: 7775
NEW: 2208
Actual rating entries: 41734


# Zomato Dataset — Data Exploration

## 1. Objective

The objective of this notebook is to understand the structure, quality, and characteristics of the historical Zomato restaurant dataset before performing data cleaning and feature engineering.

This dataset will be used as the historical business data component of the **Agentic AI-Based Business Potential Prediction in a Locality** project.

## 2. Project Context

The overall system aims to predict the potential of a food and beverage business in a given locality by combining:

- Historical business data from Zomato
- Current competition data from Google Places API
- Population and demographic data
- Locality and infrastructure data from OpenStreetMap (OSM)

In this notebook, we focus only on the **Zomato historical dataset**.

## 3. Objectives of Data Exploration

The following aspects of the dataset will be examined:

- Dataset dimensions (rows and columns)
- Column names and data types
- Missing values
- Duplicate records
- Unique values in categorical columns
- Distribution of important numerical variables
- Restaurant types and business categories
- Location distribution
- Cuisine distribution
- Rating distribution
- Cost distribution
- Potential features relevant to business performance prediction

## 4. Dataset Overview

The dataset contains information about restaurants in Bengaluru, including:

- Restaurant name and location
- Online ordering and table booking
- Restaurant ratings
- Votes/reviews
- Restaurant type
- Cuisines
- Approximate cost for two people
- Business category
- Listed city/location information

### Initial Dataset Statistics

- **Rows:** 51,717
- **Columns:** 17
- **Duplicate rows:** 0

## 5. Important Note

At this stage, no data cleaning, feature engineering, or machine learning transformations are performed.

The purpose of this notebook is only to **understand the raw dataset**.

Cleaning and preprocessing will be performed separately in:

`02_zomato_cleaning.ipynb`

## 6. Expected Outcome

By the end of this notebook, we should have a clear understanding of:

1. Which variables are useful for the project
2. Which variables contain missing or inconsistent values
3. Which categorical variables require preprocessing
4. Which numerical variables require transformation
5. Which features may contribute to the definition of business performance/success

This analysis will guide the subsequent **data cleaning, feature engineering, and machine learning stages**.